wandb/sdk/wandb_run.py

```diff
@@ -373,6 +377,9 @@ def __init__(
 
         # Initialize telemetry object
         self._telemetry_obj = telemetry.TelemetryRecord()
+        self._telemetry_obj_active = False
```

Comment:
if we have `dirty` do we need `active` as well? trying to understand this use case

Reply:
yep... active indicates whether we should be flushing the object or just caching the data.

So you can be active and dirty and not active and not dirty, etc

This is primarily needed before the Run is completely setup... the code might add a bunch of telemetry data but if we tried to send it, it might be too early and cause strange things to happen in the handler and sender.   so we wait to turn the telemetry to active until the Run is "ready".  the new _on_ state that was introduced.

Reply:
so is this `ready` flag should be general ... not just for telemetry? are there other cases we might try send things before we are "ready"?

Reply:
not yet,  so i kept it specific since there was only one use.  If there are more, we can make it more general.
Another partially related one was the _frozen state on the Run... but those were both set in the _on_ready event.

To me it is more clear thinking about the specific object actions together... otherwise you would have some logic that would be doing different things based on a more global run state and some local telemetry cache states.

Reply:
ok fair enough! 😄